### PR TITLE
fix: add remote build image building

### PR DESCRIPTION
**Pull request by jLopezbarb**
# Proposed changes

Fixes https://github.com/okteto/app/issues/6122

- Checks if the image that is in the `destroy.image` is already built and build it if it's not built

### CODE REVIEW

File: cmd/destroy/build_control.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+		}
+		svcsToBuild, err := bc.builder.GetServicesToBuild(ctx, manifest, []string{svc})
+		if err != nil {
+			return fmt.Errorf("error getting services to build: %w", err)
+		}
+		if len(svcsToBuild) == 0 {
+			oktetoLog.Infof("image is already built: %s", imageToBuild)
+			return nil
+		}
+		buildOptions := &types.BuildOptions{
```
Comment (teresaromero) [inline cmd/destroy/build_control.go:76]: are this the only options we want to pass though to build? what about image, etc?

File: cmd/destroy/destroy.go [hunk 2]
```diff
@@ -244,6 +245,16 @@ func (dc *destroyCommand) getDestroyer(opts *Options) (destroyInterface, error)
 
 		destroyImage := ""
 		if manifest.Destroy != nil {
+			if opts.Name == "" {
+				if manifest.Name == "" {
+					manifest.Name = dc.buildCtrl.name
```
Comment (teresaromero) [inline cmd/destroy/destroy.go:250]: where does `dc.builderCtrl` get `name` ?

Comment (jLopezbarb) [reply cmd/destroy/destroy.go:250]: In the initializer. in the `destroy.go`. The name is the inferred name (from the cmap or the folder name)

File: cmd/destroy/build_control.go (hunk 1 above)
Comment (jLopezbarb) [reply cmd/destroy/build_control.go:76]: We get the image information from the manifest.Build section. I'm adding `CommandArgs: svcsToBuild,` with the service that needs to be built to have the logs with the stage

**Comment by jLopezbarb** [comment]
> testing this change I notice that if we need to create the image we don't display any message related to that process. Maybe the user expects some log related to his destroy commands but we only see a 'stuck' destroy command (because is building the image). Should we add some info to the user saying that destroy image is being built?
> 
> Steps to repro:
> 
> * clone https://github.com/okteto/movies/tree/adrian/test-3493
> * exec `okteto deploy`
> * exec `okteto destroy`
> * check no info displayed related to building destroy image (it may take time)
> 
> Even a message of already built images is displayed when this is not the case.

Hi @AdrianPedriza  couldn't reproduce your issue, but checking the build there was an optimisation to be done at the image check. We were checking all the images even if we only needed one of them. Here is [a loom video](https://www.loom.com/share/0ffd7417d42a4976aded116fb8200b52) to show you that is working as expected

**Comment by AdrianPedriza** [comment]
> > testing this change I notice that if we need to create the image we don't display any message related to that process. Maybe the user expects some log related to his destroy commands but we only see a 'stuck' destroy command (because is building the image). Should we add some info to the user saying that destroy image is being built?
> > Steps to repro:
> > 
> > * clone https://github.com/okteto/movies/tree/adrian/test-3493
> > * exec `okteto deploy`
> > * exec `okteto destroy`
> > * check no info displayed related to building destroy image (it may take time)
> > 
> > Even a message of already built images is displayed when this is not the case.
> 
> Hi @AdrianPedriza couldn't reproduce your issue, but checking the build there was an optimisation to be done at the image check. We were checking all the images even if we only needed one of them. Here is [a loom video](https://www.loom.com/share/0ffd7417d42a4976aded116fb8200b52) to show you that is working as expected

nevermind, I think I tested it wrong. I have tested it again and it works as it should! :)

**Review by AdrianPedriza — APPROVED** [review]

**Review by andreafalzetti — APPROVED** [review]
LGTM

**Pull request merged**